packages/terra-table/src/TableRows.jsx

```diff
@@ -34,17 +35,35 @@ function cloneChildItems(children, onClick, onKeyDown) {
     if (child.type === TableRow) {
       return React.cloneElement(child, newProps);
     }
+    if (child.type === TableSubheader) {
+      return React.cloneElement(child, { count: numberOfCols });
+    }
     return child;
   });
 }
 
+function getNumberOfColumns(children) {
+  let count = 16;
+  const childArray = React.Children.toArray(children);
+  for (let i = 0; i < childArray.length; i += 1) {
+    // If the child is of type TableRow and it has children, then return the count of children.
```

Comment:
nit: It took me a while to realize what you were doing when I read this comment.  Consider rewording?